sys/kern/kern_proc.c

```diff
@@ -3175,18 +3212,38 @@ sysctl_kern_proc_sigtramp(SYSCTL_HANDLER_ARGS)
 		error = SYSCTL_OUT(req, &kst32, sizeof(kst32));
 		return (error);
 	}
+#endif
+#ifdef COMPAT_FREEBSD64
+	if ((req->flags & SCTL_MASK64) != 0) {
+		bzero(&kst64, sizeof(kst64));
+		if (!SV_PROC_FLAG(p, SV_CHERI)) {
+			if (sv->sv_sigcode_base != 0) {
+				kst64.ksigtramp_start = sv->sv_sigcode_base;
+				kst64.ksigtramp_end = sv->sv_sigcode_base +
+				    *sv->sv_szsigcode;
+			} else {
+				kst64.ksigtramp_start = sv->sv_psstrings -
+				    *sv->sv_szsigcode;
+				kst64.ksigtramp_end = sv->sv_psstrings;
+			}
+		}
+		PROC_UNLOCK(p);
+		error = SYSCTL_OUT(req, &kst64, sizeof(kst64));
+		return (error);
+	}
 #endif
 	bzero(&kst, sizeof(kst));
 	if (sv->sv_sigcode_base != 0) {
-		kst.ksigtramp_start = (char *)sv->sv_sigcode_base;
-		kst.ksigtramp_end = (char *)sv->sv_sigcode_base +
-		    *sv->sv_szsigcode;
+		kst.ksigtramp_start = EXPORT_KPTR((char *)sv->sv_sigcode_base);
+		kst.ksigtramp_end = EXPORT_KPTR((char *)sv->sv_sigcode_base +
+		    *sv->sv_szsigcode);
 	} else {
-		kst.ksigtramp_start = (char *)sv->sv_psstrings -
-		    *sv->sv_szsigcode;
-		kst.ksigtramp_end = (char *)sv->sv_psstrings;
+		kst.ksigtramp_start = EXPORT_KPTR((char *)sv->sv_psstrings -
+		    *sv->sv_szsigcode);
+		kst.ksigtramp_end = EXPORT_KPTR((char *)sv->sv_psstrings);
 	}
 	PROC_UNLOCK(p);
+	/* XXX-CHERI: this strips tags... */
```

Comment:
I think that's ok for the purposes of this sysctl, so I would maybe tweak it `NB CHERI: This strips tags.`  In particular, it would only ever be safe to not strip tags for curproc, but even then it seems of dubious value to return a capability to arbitrary user space code.